File: 9.palindrome-number.py
#
# @lc app=leetcode id=9 lang=python3
#
# [9] Palindrome Number
#

# @lc code=start


class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0 :
            return False
        x = str(x)

        if x == x[::-1]:
            return True
        return False
    
        
# @lc code=end

File: 9.palindrome-number.rs
/*
 * @lc app=leetcode id=9 lang=rust
 *
 * [9] Palindrome Number
 */

// @lc code=start
impl Solution {
    pub fn is_palindrome(x: i32) -> bool {
        let x_string : String = x.to_string();
        let x_string_rev: String = x_string.chars().rev().collect();

        if x_string == x_string_rev {
            return true;
        } else {
            return false;
        }
    }
}
// @lc code=end

